src/gadgets/crt/mod.rs

```diff
@@ -337,8 +323,10 @@ mod tests {
         // Perform subtraction
         let crt_diff = crt_poly_a.sub(&crt_poly_b, &mut circuit);
 
-        // Finalize to get output
-        let outputs = crt_diff.finalize(&mut circuit);
+        // Finalize per-slot contributions and full CRT reconstruction
+        let mut outputs = crt_diff.finalize_crt(&mut circuit);
```

Comment:
can we put `finalize_crt` as internal call for `finalize_reconst`? does it have to be seperately called?